api/api.go

```diff
@@ -29,6 +37,25 @@ type Error struct {
 	// be valid or invalid depending on the current state of a module.
 }
 
+type PaginationRequest struct {
+	HasPaginationQuery 			bool
+	PaginationQueryIsValid 		bool
+	Start 						int
+	Limit 						int
+}
```

Comment:
need to run `go fmt` here. If you don't run `go fmt` then the build will always fail.

Best thing to do though is to run `make test-long` before you push code, because that'll do all the formatting, linting, vetting, and of course testing for you. Takes about 10 minutes on a desktop w/SSD.

Reply:
I believe that typically we parse requests inside the function, without having a request struct to funnel code into.